### PR TITLE
Autosuggest: Add 'filterSuggestions' function

### PR DESCRIPTION
**Pull request by markdalgleish**
See changeset for details.

### CODE REVIEW

**Comment by markdalgleish** [comment]
Just realised this is missing a changeset for the Autosuggest API change. I'll get that added.

File: .changeset/cold-bugs-chew.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+**Autosuggest:** Add `filterSuggestions` function, allow `suggestions` prop to be a function
+
+The logic for filtering suggestions typically lives on the server rather than the client because it’s impractical to send all possible suggestions over the network. However, when prototyping in Playroom or working with smaller datasets, you may want to perform this filtering on the client instead. For this case, we now provide a `filterSuggestions` function to make this as painless as possible.
+
+To better support this behaviour, you can now pass a function to the `suggestions` prop. When executed, this function will be passed the current `value` of the field.
+
+**EXAMPLE USAGE**
+
+```tsx
+<Autosuggest
```
Comment (michaeltaranto) [inline .changeset/cold-bugs-chew.md:19]: What are your thoughts on showing the import here for `filterSuggestions` to drive home the possible app usage for simple datasets?

File: generate-component-docs/__snapshots__/contract.test.ts.snap [hunk 2]
```diff
@@ -6435,6 +6437,46 @@ Object {
...
+    sort: (compareFn?: ((a: Pick<Suggestion<Value>, "value" | "text" | "label" | "description" | "onClear" | "clearLabel"> | FilterableGroupedSuggestions<Value>, b: Pick<...> | FilterableGroupedSuggestions<...>) => number) | undefined) => FilterableSuggestions<...>
+    splice: { (start: number, deleteCount?: number | undefined): (Pick<Suggestion<Value>, "value" | "text" | "label" | "description" | "onClear" | "clearLabel"> | FilterableGroupedSuggestions<...>)[]; (start: number, deleteCount: number, ...items: (Pick<...> | FilterableGroupedSuggestions<...>)[]): (Pick<...> | FilterableGroupe...
+    toLocaleString: () => string
+    toString: () => string
+    unshift: (...items: (Pick<Suggestion<Value>, "value" | "text" | "label" | "description" | "onClear" | "clearLabel"> | FilterableGroupedSuggestions<Value>)[]) => number
+    values: () => IterableIterator<Pick<Suggestion<Value>, "value" | "text" | "label" | "description" | "onClear" | "clearLabel"> | FilterableGroupedSuggestions<Value>>
+},
+}
+`;
+
```
Comment (michaeltaranto) [inline generate-component-docs/__snapshots__/contract.test.ts.snap:6479]: 😫

File: lib/components/Autosuggest/filterSuggestions.test.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+      ),
+    ).toEqual([
+      {
+        text: 'Apples',
+        highlights: [{ start: 0, end: 3 }],
+      },
+    ]);
+  });
+
+  it('supports grouped lists', () => {
```
Comment (michaeltaranto) [inline lib/components/Autosuggest/filterSuggestions.test.ts:103]: Be good to tweak this or add a test that includes suggestions from both groups, e.g value equals `B`

**Review by michaeltaranto — APPROVED** [review]
Couple of minor questions/points. Great stuff 🚀

**Pull request merged**